internal/graph/taskrunner.go

```diff
@@ -0,0 +1,134 @@
+package graph
+
+import (
+	"context"
+	"sync"
+)
+
+type token struct{}
```

Comment:
nit: maybe `workerReleased` instead?

Reply:
I honestly don't know a great name for it; I just copied `token` from the errgroup impl